Content.Server/Connection/ConnectionManager.IPIntel.cs

```diff
@@ -0,0 +1,346 @@
+        return _rejectUnknown ? (true, Loc.GetString("ipintel-unknown")) : (false, string.Empty);
+    }
+
+    private bool ErrorCheck(string response, out (bool, string) result)
+    {
+        result = default;
+
+        var errorMessages = new Dictionary<string, string>
+        {
+            ["-1"] = "Invalid/No input.",
+            ["-2"] = "Invalid IP address.",
+            ["-3"] = "Unroutable address / private address given to the api. Make an issue in upstream as it should have been handled.",
+            ["-4"] = "Unable to reach IPIntel database. Perhaps it's down?",
+            ["-5"] = "Server's IP/Contact may have been banned, go to getipintel.net and make contact to be unbanned.",
+            ["-6"] = "You did not provide any contact information with your query or the contact information is invalid.",
+        };
+
+        if (!errorMessages.TryGetValue(response, out var errorMessage))
+            return false;
+
+        _sawmill.Error($"IPIntel returned error {response}: {errorMessage}");
+        result = _rejectUnknown ? (true, Loc.GetString("ipintel-unknown")) : (false, string.Empty);
+
+        return true;
+    }
+
+    private void HandleRatelimit()
+    {
+        // Oh my god this is terrible
+        if (_currentRequestsDay >= _requestLimitDay)
+        {
+            if (ShouldLiftRateLimit(_ratelimitedDay, _lastRatelimited, TimeSpan.FromDays(1)))
+            {
+                _sawmill.Info("IPIntel daily rate limit lifted. We are back to normal.");
+                _ratelimitedDay = false;
+                _currentRequestsDay = 0;
+                _limitHasBeenHandled = false;
+                return;
+            }
+
+            if (_limitHasBeenHandled)
+                return;
+
+            _sawmill.Warning($"We just hit our last daily IPIntel limit ({_requestLimitDay})");
+            _ratelimitedDay = true;
+            _lastRatelimited = _gameTiming.RealTime;
+        }
+        else if (_currentRequestsMinute >= _requestLimitMinute)
+        {
+            if (ShouldLiftRateLimit(_ratelimitedMinute, _lastRatelimited, TimeSpan.FromMinutes(1)))
+            {
+                _sawmill.Info("IPIntel minute rate limit lifted. We are back to normal.");
+                _ratelimitedMinute = false;
+                _currentRequestsMinute = 0;
+                _limitHasBeenHandled = false;
+                return;
+            }
+
+            if (_limitHasBeenHandled)
+                return;
+
+            _sawmill.Warning($"We just hit our last minute IPIntel limit ({_requestLimitMinute}).");
+            _ratelimitedMinute = true;
+            _lastRatelimited = _gameTiming.RealTime;
+        }
+    }
+
+    private bool ShouldLiftRateLimit(bool currentlyRatelimited, TimeSpan lastRatelimited, TimeSpan liftingTime)
+    {
+        // Should we raise this limit now?
+        return currentlyRatelimited &&  lastRatelimited >= liftingTime;
```

Comment:
You're missing some comparison to `IGameTiming.RealTime` here, `liftingTime` is always going to be smaller than `lastRatelimited` for minutes.